planner/cascades/implementation_rules.go

```diff
@@ -151,22 +155,71 @@ func (r *ImplTiKVSingleReadGather) OnImplement(expr *memo.GroupExpr, reqProp *pr
 	return impl.NewTableReaderImpl(reader, sg.Source.TblColHists), nil
 }
 
+// ImplTiKVDoubleReadGather implements TiKVDoubleGather as PhysicalIndexLookUpReader.
+type ImplTiKVDoubleReadGather struct {
+}
+
+// Match implements ImplementationRule Match interface.
+func (r *ImplTiKVDoubleReadGather) Match(expr *memo.GroupExpr, prop *property.PhysicalProperty) (matched bool) {
+	if prop.IsEmpty() {
+		return true
+	}
+	reader := expr.ExprNode.(*plannercore.TiKVDoubleGather)
+	if all, _ := prop.AllSameOrder(); all {
+		for i, col := range reader.IndexCols {
```

Comment:
Don't we need to consider the where equal condition ends?

Reply:
+1, have you added such case?

Reply:
For now, double read can't push down any conditions. And the `reader.IndexCols` is the index columns involved in the schema. So I think there are no problems here.

Reply:
Even if there is no conditions push down, it is the same as equal conditions ends at 0, then it still have problems.

Reply:
I have changed the `Match` function to directly return `true`. And let the property deferred checked by `ImplIndexScan`.